app/src/org/commcare/views/widgets/AudioPrototype.java

```diff
@@ -0,0 +1,211 @@
+        }
+    }
+
+    @Override
+    public void onRecordingCompletion(){
+        setBinaryData(recorder.getFileName());
+        mPlayButton.setEnabled(true);
+        mPlayButton.setBackgroundResource(R.drawable.play);
+        recordingNameText.setTextColor(getResources().getColor(R.color.black));
+        recordingNameText.setText(Localization.get("recording.custom"));
+    }
+
+    @Override
+    protected void playAudio() {
+        Uri myPath = Uri.parse(mInstanceFolder + mBinaryName);
+        final MediaPlayer player = MediaPlayer.create(getContext(), myPath);
+        player.setOnCompletionListener(new MediaPlayer.OnCompletionListener() {
+            @Override
+            public void onCompletion(MediaPlayer mp) {
+                resetAudioPlayer(mp);
+            }
+        });
+        player.start();
+        mPlayButton.setBackgroundResource(R.drawable.pause);
+        mPlayButton.setOnClickListener(new OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                GoogleAnalyticsUtils.reportAudioPaused();
+                pauseAudioPlayer(player);
+            }
+        });
+    }
+
+    private void pauseAudioPlayer(MediaPlayer player){
+        final MediaPlayer mp = player;
+        mp.pause();
+        mPlayButton.setBackgroundResource(R.drawable.play);
+        mPlayButton.setOnClickListener(new OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                GoogleAnalyticsUtils.reportAudioPlayed();
+                resumeAudioPlayer(mp);
+            }
+        });
+    }
+
+    private void resumeAudioPlayer(MediaPlayer player){
+        final MediaPlayer mp = player;
+        mp.start();
+        mPlayButton.setBackgroundResource(R.drawable.pause);
+        mPlayButton.setOnClickListener(new OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                GoogleAnalyticsUtils.reportAudioPaused();
+                pauseAudioPlayer(mp);
+            }
+        });
+    }
+
+    private void resetAudioPlayer(MediaPlayer mp){
+        mp.release();
+        mPlayButton.setBackgroundResource(R.drawable.play);
+        mPlayButton.setOnClickListener(new OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                GoogleAnalyticsUtils.reportAudioPlayed();
+                playAudio();
+            }
+        });
+    }
+
+    @Override
+    protected void togglePlayButton(boolean enabled) {
+        if(enabled){
+            mPlayButton.setBackgroundResource(R.drawable.play);
+        }else{
+            mPlayButton.setBackgroundResource(R.drawable.play_disabled);
+        }
+        mPlayButton.setEnabled(enabled);
+    }
+
+    @Override
+    protected void reloadFile(){
+        super.reloadFile();
+        recordingNameText.setTextColor(getResources().getColor(R.color.black));
+        if(mBinaryName.contains(CUSTOM_TAG)){

```

Comment:
@wpride we want to display "Custom Recording" if they load a file recorded in the widget, and the file name if they choose a file from the device. 
This is checking if it's a custom recording.